src/Npgsql/PostgresException.cs

```diff
@@ -236,6 +236,40 @@ internal PostgresException(ReadBuffer buf)
         public override string Message => SqlState + ": " + MessageText;
 
         /// <summary>
+        /// Gets whether or not the exception can be treated as transient based on the SqlState. Useful when defining retry policies.
+        /// </summary>
+        public override bool IsTransient
+        {
+            get
+            {
+                switch (SqlState)
+                {   
+                    case "53000":   //insufficient_resources
+                    case "53100":   //disk_full
+                    case "53200":   //out_of_memory
+                    case "53300":   //too_many_connections
+                    case "53400":   //configuration_limit_exceeded
+                    case "57P03":   //cannot_connect_now
+                    case "58000":   //system_error
+                    case "58030":   //io_error
+                    case "40001":   //serialization_error
+                    case "55P03":   //lock_not_available
```

Comment:
Yeah, but it seems that like Craig says, it depends on your definition of "transient". I'm not convinced that's the kind of transient we're looking for. Specifically for 55P03 and 55006, it would be good to understand a bit more when they're thrown.

If you don't have any time to dig into this, we can merge with these errors - as we said before the worse case is that we retry for nothing.